tests/benchmark/utils.py

```diff
@@ -0,0 +1,129 @@
+"""Google Drive utils."""
+
+import io
+import json
+import os
+import pathlib
+import sys
+import tempfile
+from functools import lru_cache
+
+import pandas as pd
+import yaml
+from pydrive.auth import GoogleAuth
+from pydrive.drive import GoogleDrive
+
+PYDRIVE_CREDENTIALS = 'PYDRIVE_CREDENTIALS'
+BENCHMARK_FILE_ID = '1mrvIok6G5P0x88m2_TjOtqcQ-p4iEAk8PTuG6Hpp5Uk'
+
+
+def detect_python_version():
+    """Detect python version being used."""
+    python_version = sys.version_info
+    python_version = f'{python_version.major}.{python_version.minor}'
+    return python_version
+
+
+def get_previous_result(dtype, method):
+    """Return previous result for a given ``dtype`` and method."""
+    data = read_excel(BENCHMARK_FILE_ID)
+    python_version = detect_python_version()
+    df = data[python_version]
+    filtered_row = df[df['dtype'] == dtype]
+    value = filtered_row[method].to_numpy()[0]
+    return value
+
+
+def _get_drive_client():
```

Comment:
can we actually pull all the gdrive stuff into a higher level module? Like tests/_external/google_utils or something like that?